core/dbt/task/runnable.py

```diff
@@ -320,3 +333,73 @@ def get_result(self, results, elapsed_time, generated_at):
 
     def task_end_messages(self, results):
         dbt.ui.printer.print_run_end_messages(results)
+
+
+class RemoteCallable(object):
+    METHOD_NAME = None
+    is_async = False
+
+    @abstractmethod
+    def handle_request(self, **kwargs):
+        raise dbt.exceptions.NotImplementedException(
+            'from_kwargs not implemented'
+        )
+
+    def _subprocess_handle_request(self, conn, **kwargs):
+        error = None
+        result = None
+        try:
+            result = self.handle_request(**kwargs)
+        except dbt.exceptions.RuntimeException as exc:
+            logger.debug('dbt runtime exception',
+                         exc_info=True)
+            # we have to convert this to a string for RPC responses
+            error = str(exc)
+        except dbt.exceptions.RPCException as exc:
+            error = str(exc)
+        except Exception as exc:
+            logger.debug('uncaught python exception',
+                         exc_info=True)
+            error = str(exc)
+        conn.send([result, error])
+
+    def safe_handle_request(self, **kwargs):
```

Comment:
How happy are you with this implementation of timeouts? It doesn't look unreasonable to me, but curious if you view this as something close to production-ready, or if your TODO above indicates that we need to give this some more thought

Reply:
I think this this is probably how it will shake out, I'm fine with it, I think. I mostly anticipate issues on Windows, potentially.

I do want to make sure we're `close()`ing everything that needs it though.